platform/viewer/src/appExtensions/MeasurementsPanel/ConnectedMeasurementTable.js

```diff
@@ -201,26 +198,20 @@ const mapDispatchToProps = dispatch => {
         return;
       }
 
-      const { element } = enabledElements[activeViewportIndex];
-
-      const eventData = {
-        event: {
-          clientX: event.clientX,
-          clientY: event.clientY,
-        },
-        element,
-      };
-
       const { toolType, measurementId } = measurementData;
       const tool = MeasurementApi.Instance.tools[toolType].find(measurement => {
         return measurement._id === measurementId;
       });
 
-      const options = {
+      const props = {
         editDescriptionOnDialog: true,
+        measurementData: tool,
       };
 
-      getMeasurementLocationCallback(eventData, tool, options);
+      ownProps.commandsManager.runCommand('showLabellingFlow', {
```

Comment:
Similar to other things we've discussed. I'm hesitant to tie a component directly to `commandsManager`
Because this would be isolated to an extension, that bothers me less, but still something we should discuss.

Reply:
Totally agree. Done! thanks.